drake/systems/plants/@RigidBodyManipulator/doKinematics.m

```diff
@@ -99,6 +108,20 @@
   else
     kinsol.mex_ptr = options.kinematics_cache_ptr_to_use;
   end
+
+  if options.compute_gradients
+    nq = length(q);
+    nv = length(v);
+    if isnumeric(q)
+      q = TaylorVar(q, {[eye(nq), zeros(nq, nv)]});
```

Comment:
This seems potentially inefficient; but I guess it keeps the mex interface abstraction clean?